### ISSUE TITLE
Fow indexed access types results in `unknow`

### ISSUE THREAD

**Issue by pascalduez** [reactjs/react-docgen#837]
Hi,

I guess this is similar to #367, #369 for TS.

Using an indexed access type for a component prop result in `unknow`. Whereas it should result in `Type['index']`.   
Either using types or interfaces.   

```js
// @flow 

type Foo = {
  name: string
}

type Props = {
  name: Foo['name']
}

// @component
function Component(props: Props) {}
```

```json
[
  {
    "description": "",
    "displayName": "Component",
    "methods": [],
    "props": {
      "name": {
        "required": true,
        "flowType": {
          "name": "unknown"
        },
        "description": ""
      }
    }
  }
]
```

**Comment by pascalduez** [comment]
That was fast, thanks @danez!

**Comment by danez** [comment]
This will be available in 6.0.4
